File: eye-recognition-serivce/model/EyeRecognitionModel.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime
from .EyeRecognitionSampleHistory import EyeRecognitionSampleHistory
import uuid

class EyeRecognitionModel(BaseModel):
    id: Optional[str] = None
    modelLink: str = None
    eyeModelName: str  = None
    eyeRecognitionSampleHistory: list[EyeRecognitionSampleHistory]  = None
    accuracy: Optional[float] = None
    f1Score: Optional[float] = None
    precision: Optional[float] = None
    isActive: bool = True
    epochs: Optional[int] = None
    learningRate: Optional[float] = None
    imageSize: Optional[int] = None
    batchSize: Optional[int] = None
    mappingLabel: Optional[str] = None
    trainingTime:Optional[int] = None
    createDate: Optional[datetime] = None
   
    class Config:
        from_attributes = True

File: backend-server/src/main/java/com/memberservice/entity/EyeRecognitionModel.java
package com.memberservice.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.GenericGenerator;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "tblEyeRecognitionModel")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class EyeRecognitionModel {

    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    @Column(name = "id", updatable = false, nullable = false)
    private UUID id;

    @Column(name = "modelLink")
    private String modelLink;

    @Column(name = "eyeModelName")
    private String eyeModelName;

    @Column(name = "accuracy")
    private Double accuracy;

    @Column(name = "f1Score")
    private Double f1Score;

    @Column(name = "precision")
    private Double precision;

    @Column(name = "isActive", nullable = false)
    private Boolean isActive = true;

    @Column(name = "epochs")
    private Integer epochs;

    @Column(name = "learningRate")
    private Double learningRate;

    @Column(name = "imageSize")
    private Integer imageSize;

    @Column(name = "batchSize")
    private Integer batchSize;

    @Column(name = "mappingLabel")
    private String mappingLabel;

    @Column(name = "trainingTime")
    private Integer trainingTime;

    @Column(name = "createDate")
    private LocalDateTime createDate;

    @Column(name = "modeltype")
    private String modelType;

    @OneToMany(mappedBy = "model", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JsonProperty("eyeRecognitionSampleHistory")
    private List<EyeRecognitionSampleHistory> histories;
} 